src/libstore/build.cc

```diff
@@ -530,6 +532,12 @@ class UserLock
 UserLock::UserLock()
 {
     assert(settings.buildUsersGroup != "");
+    createDirs(settings.nixStateDir + "/userpool");
+    /* Mark that user is not enabled by default */
+    uid = 0;
```

Comment:
`0` seems a risky choice as a sentinel value for UIDs given that 0 == root. So if there's ever any confusion, we might end up doing an operation as/to root. Might be better to use `-1` or `std::optional<uid_t>`.

Reply:
Alternatively, to ensure that `UserLock` always represents a valid user, have a builder function that returns `std::unique_ptr<UserLock>` so it can represent failure to acquire a user.

Reply:
I've thought about adding a function that can represent a failure, and it does appear to be a better solution from some perspectives, but I didn't want to make the diff too big (since this is my first PR). Now that it's inevitably going to be massive, I'll wrap the result in optional and make it represent failure.